.github/workflows/release.yml

```diff
@@ -0,0 +1,65 @@
+name: Create release
+on: [push]
+
+jobs:
+  release:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - uses: actions/setup-python@v5
+      with:
+        python-version: "3.8"
+
+    - name: Install Hatch
+      run: pipx install hatch
+
+    - name: Set variables
+      id: set_variables
+      run: |
+        tag="$(git describe --tags --abbrev=0)"
```

Comment:
This seems to assume a git tag already exists

Reply:
Right I forgot I tested with a failed tag existing

Reply:
```bash
tag=$(git for-each-ref refs/tags --sort=-taggerdate --format='%(refname:strip=2)' --count=1)
```
This gives an empty string when no tags are present. It works without problem creating the first release of the repo.
Refer to https://stackoverflow.com/a/5261470